File: src/first_package/service/add_number_client.cpp
#include "ros/ros.h"
#include "first_package/add_number.h"
#include <cstdlib>

int main(int argc, char **argv)
{
    //init the client node
    ros::init(argc, argv, "add_two_number_client");
    if (argc != 3)
  {
    ROS_INFO("usage: add_two_number_client X Y");
    return 1;
  }

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<first_package::add_number>("add_two_number");
  first_package::add_number srv;
  srv.request.x = atoll(argv[1]);
  srv.request.y = atoll(argv[2]);
  if (client.call(srv))
  {
    ROS_INFO("Sum: %ld", (long int)srv.response.total);
  }
  else
  {
    ROS_ERROR("Failed to call service add_two_number");
    return 1;
  }

  return 0;
}

File: src/first_package/service/add_number_client.py
#!/usr/bin/env python

import sys
import rospy
from first_package.srv import add_number
from first_package.srv import add_numberRequest
from first_package.srv import add_numberResponse

def add_two_ints_client(x, y):
    #wait for the server add_two_number alive
    rospy.wait_for_service('add_two_number')
    try:
        #create service proxy that include (server namem server type) sending the information to the server
        add_two_ints = rospy.ServiceProxy('add_two_number', add_number)
        #call into the server that can add two number
        resp1 = add_two_ints(x, y)
        #return the total number
        return resp1.total
    except rospy.ServiceException(e):
        print ("Service call failed: %s"%e)

def usage():
    return 

if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
    else:
        print ("%s [x y]"%sys.argv[0])
        sys.exit(1)
    print ("Requesting %s+%s"%(x, y))
    s = add_two_ints_client(x, y)
    print ("%s + %s = %s"%(x, y, s))